cmd/picoshare/main.go

```diff
@@ -75,6 +76,13 @@ func main() {
 
 func requireEnv(key string) string {
 	val := os.Getenv(key)
+	if val == "" {
```

Comment:
We can do whatever you decide: it's your project :smile: 

Picoshare currently only calls `requireEnv` once: for `PS_SHARED_SECRET`. This change, in the current code, would not modify any other behavior. If a second invocation of `requireEnv` is added later, it will provide consistency for users who want it.

Your counter-proposal will certainly achieve the same end, and I have used this same technique for other packages that don't easily support Docker's secrets mechanism. If this is the way you'd like to go, I recommend instead bind-mounting a new entrypoint script that sets environment variables if the corresponding `_FILE` variable is set, so people don't have to worry about escaping rules for special characters. But I feel like picoshare would be improved by providing this idiomatic functionality, which is why I proposed the merge.

Would you feel better about this addition to `docker-entrypoint`?

```sh
[ -z "$PS_SHARED_SECRET_FILE" ] || PS_SHARED_SECRET=$(cat $PS_SHARED_SECRET_FILE)
```

Reply:
Yeah, I'd like to limit it to just a doc change without changing the code.

How about we update the current [Docker Compose section](https://github.com/mtlynch/picoshare#using-docker-compose) to explain how to do it with a secrets file?

Reply:
All right, that would be 0 lines of code authored by me.

Would you like me to take this merge request out of draft before you reject it?

Reply:
Oh, I thought you'd update this one with the new implementation. I've closed this one. Feel free to send a doc update PR!